lms/djangoapps/courseware/access.py

```diff
@@ -325,10 +351,10 @@ def can_see_in_catalog():
         In this case we use the catalog_visibility property on the course descriptor
         but also allow course staff to see this.
         """
-        return (
+        return ACCESS_GRANTED if (
             course.catalog_visibility == CATALOG_VISIBILITY_CATALOG_AND_ABOUT or
```

Comment:
to be consistent, create a `_has_catalog_visibility` helper function that takes in the visibility type.